### PR TITLE
Fix sidecar index out of range issues

### PR DESCRIPTION
**Pull request by mbhall88**
Fixes the issues mentioned in #96 and #113. Used the patch from https://github.com/Snakemake-Profiles/slurm/issues/96#issuecomment-1429260589

### CODE REVIEW

**Comment by percyfal** [comment]
Thanks for the PR @mbhall88! I had just started looking into this but not added any code yet. I was thinking of adding a test, but if the only case this happens is for unseen job ids that may be unnecessary. 

I just added some comments regarding the use of `--me` option, which is unavailable for older slurm versions (#101).

**Comment by mbhall88** [comment]
@percyfal I don't see any of the comments you mentioned?

**Comment by percyfal** [comment]
> @percyfal I don't see any of the comments you mentioned?

I guess pressing "Submit review" helps... Better now?

File: {{cookiecutter.profile_name}}/slurm-sidecar.py [hunk 1]
```diff
@@ -1,9 +1,9 @@
 #!/usr/bin/env python3
 """Run a Snakemake v7+ sidecar process for Slurm
 
-This sidecar process will poll ``squeue --user [user] --format='%i,%T'``
-every 60 seconds by default (use environment variable
-``SNAKEMAKE_SLURM_SQUEUE_WAIT`` for adjusting this).
+This sidecar process will poll ``squeue --me --format='%i,%T'`` every 60
```
[the remark this answers is not in the document]
Comment (mbhall88) [reply {{cookiecutter.profile_name}}/slurm-sidecar.py]: Sorry some of the patch changes were from older versions. I have now tried to just select the changes relevant to the issue

File: {{cookiecutter.profile_name}}/slurm-sidecar.py [hunk 2]
```diff
@@ -209,6 +217,12 @@ def do_GET(self):
             return
         # Otherwise, query job ID status
         job_id = self.path[len("/job/status/") :]
+        try:
+            logger.debug("CLEMENS:")
```
[the remark this answers is not in the document]
Comment (mbhall88) [reply {{cookiecutter.profile_name}}/slurm-sidecar.py]: I don;t know. This was in the patch also. I've removed it now

**Pull request merged**